hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/infra/HevmTransactionFactory.java

```diff
@@ -326,8 +326,8 @@ private void assertValidCreation(@NonNull final ContractCreateTransactionBody bo
         validateTrue(body.gas() <= contractsConfig.maxGasPerSec(), MAX_GAS_LIMIT_EXCEEDED);
         final var usesUnsupportedAutoAssociations =
                 body.maxAutomaticTokenAssociations() > 0 && !contractsConfig.allowAutoAssociations();
-        final var usesInvalidAutoAssociations = body.maxAutomaticTokenAssociations() < UNLIMITED_AUTOMATIC_ASSOCIATIONS
-                && entitiesConfig.unlimitedAutoAssociationsEnabled();
+        final var usesInvalidAutoAssociations = body.maxAutomaticTokenAssociations()
+                < (entitiesConfig.unlimitedAutoAssociationsEnabled() ? UNLIMITED_AUTOMATIC_ASSOCIATIONS : 0);
```

Comment:
Is this encoding the knowledge that `.maxAutomaticTokenAssociations` must be `-1` for unlimited?  It'd be kind of nicer to be explicit.  IMO.